src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,190 @@
+                    break;
+                }
+            } else {
+                ui.print(taskList.getList());
+            }
+            return 0;
+        case "mark":
+            taskList.getTask(Integer.parseInt(args[1]) - 1).mark();
+            ui.print("Task marked as done: ", " " + taskList.getTask(Integer.parseInt(args[1]) - 1).toString());
+            return 1;
+        case "unmark":
+            taskList.getTask(Integer.parseInt(args[1]) - 1).unmark();
+            ui.print("Task as not done yet: ", " " + taskList.getTask(Integer.parseInt(args[1]) - 1).toString());
+            return 1;
+        case "todo":
+            taskList.addTask((inputString).substring(5).trim(), false, null, 0);
+            ui.print("Added Task: ", " " + taskList.getLast().toString(),
+                    String.format("There are now %d task(s) in the list.", taskList.getSize()));
+            return 1;
+        case "deadline":
+            taskList.addTask(inputString.substring(9).split("/by")[0].trim(), false,
+                    LocalDate.parse(inputString.split("/by")[1].substring(1), formatter), 1);
+            ui.print("Added Task: ", " " + taskList.getLast().toString(),
+                    String.format("There are now %d task(s) in the list.", taskList.getSize()));
+            return 1;
+        case "event":
+            taskList.addTask(inputString.substring(6).split("/at")[0].trim(), false,
+                    LocalDate.parse(inputString.split("/at")[1].substring(1), formatter), 2);
+            ui.print("Added Task: ", " " + taskList.getLast().toString(),
+                    String.format("There are now %d task(s) in the list.", taskList.getSize()));
+            return 1;
+        case "delete":
+            String deletedTaskString = taskList.getTask(Integer.parseInt(args[1]) - 1).toString();
+            taskList.removeTask(Integer.parseInt(args[1]) - 1);
+            ui.print("Deleted Task:", " " + deletedTaskString,
+                    String.format("There are now %d task(s) in the list.", taskList.getSize()));
+            return 1;
+        case "find":
+            ui.print(taskList.findTasksContaining(args[1]), "Here are the matching tasks in your list:");
+            return 1;
+        default:
+            System.out.println("Unknown Command");
+            return 0;
+        }
+    }
+
+    /**
+     * Validates a given String and argument list based on the given action and task list. <br>
+     * A DukeException is thrown when the String or argument do not fit the corresponding <br>
+     * action and task list provided. <br>
+     * <br>
+     *
+     * @param inputString The string to be checked.
+     * @param action The type action to be checked with.
+     * @param args The argument list to be checked.
+     * @param taskList The list of task to be checked with.
+     */
+
+    public void validate(String inputString, String action, String[] args, TaskList taskList) throws DukeException {
```

Comment:
I like how you encapsulate the validation logic into a single function. Consider however, making it more clear what exactly you are validating.
```suggestion
    public void validateArguments(String inputString, String action, String[] args, TaskList taskList) throws DukeException {
```